### PR TITLE
Bugfixes

### LINKED ISSUE #165: [PE-D] addpatient command error message not specific
Input:
addpatient n/test6 t/36.0 d/12345678-12345678 p/1234567 a/12

Date format is valid but the error message is:
Period of stay should be in the format YYYYMMDD-YYYYMMDD.

I'm guessing that this is because the date is not a valid one. But it doesn't quite say that it has to be a valid date in the UG.

-------------
Labels: `severity.Low` `type.FunctionalityBug`
original: KishenKumarrrrr/ped#1

### LINKED ISSUE #166: [PE-D] addpatient unable to add single day stay
Input:
addpatient n/Betsy Crowe t/36.5 d/20201001-20201001 p/91234567 a/19 c/Is asthmatic

Gives me an invalid date format but my date is valid. Doesn’t allow the person to stay for one day only which I feel is a big problem. 

-------------
Labels: `severity.High` `type.FunctionalityBug`
original: KishenKumarrrrr/ped#2

### LINKED ISSUE #167: [PE-D] addpatient command empty comment.
input:
addpatient n/123123 t/36.5 d/20201001-20201014 p/91234567 a/19 c/(no blank space)

When the comment prefix is supplied but no comment is actually given as input, it still takes it as valid input and then leaves the display blank.

-------------
Labels: `severity.Low` `type.FunctionalityBug`
original: KishenKumarrrrr/ped#3

### LINKED ISSUE #168: [PE-D] addpatient incorrect error message
input :
addpatient n/123124 t/36.5 d/20201001-20201014 p/91234567 a/120 c/something

Error message says age should only contain numbers. The issue here is that I am out of the valid range stated by the UG so I feel that the error message given should reflect that instead.

-------------
Labels: `severity.Low` `type.FunctionalityBug`
original: KishenKumarrrrr/ped#4

### LINKED ISSUE #170: [PE-D] addpatient command incorrect error message for temperature
input:
addpatient n/testt3 t/0.0 d/20201001-20201014 p/12313123 a/15

I get the error message "Temperature should only contain numbers, and it should be to one decimal place. (E.g. 36.0)" but then according to the user guide my input should be valid since no range was given. 
Potentially v troubling because now I dont know what the actual accepted range is and it could affect the usage of this application.

-------------
Labels: `severity.Medium` `type.FunctionalityBug`
original: KishenKumarrrrr/ped#6

### LINKED ISSUE #171: [PE-D] Name run-off when supplying long names
input : 
addpatient n/SUPERLONGNAMESUPERLONGNAMESUPERLONGNAMESUPERLONGNAMESUPERLONGNAMESUPERLONGNAMESUPERLONGNAMESUPERLONGNAMESUPERLONGNAMESUPERLONGNAMESUPERLONGNAMESUPERLONGNAMESUPERLONGNAMESUPERLONGNAMESUPERLONGNAMESUPERLONGNAME t/36.5 d/20201001-20201014 p/12313123 a/15

Name runs off the screen. No max name length was given in the UG.

-------------
Labels: `severity.Low` `type.FunctionalityBug`
original: KishenKumarrrrr/ped#7

### LINKED ISSUE #183: [PE-D] Error message for PERIOD_OF_STAY can be more specifc
Entering the following command 'addpatient n/John Doe t/27.0 d/20200930-20200920 p/98765432 a/35' will result in the error message 'Period of stay should be in the format YYYYMMDD-YYYYMMDD.' to be shown. This might be a bit misleading since the format is correct, just that the start date is more than end date. Since the code is already checking for valid start dates and end dates, maybe you guys could throw another error message for it?

-------------
Labels: `severity.Low` `type.FeatureFlaw`
original: yuxuanxc/ped#1

### LINKED ISSUE #184: [PE-D] Duration of PERIOD_OF_STAY is not mentioned in User guide
It was not mentioned in the user guide that the period of stay must be more than a day, but if the user inputs 'addpatient n/John Doe t/27.0 d/20200920-20200920 p/98765432 a/35' the error message 'Period of stay should be in the format YYYYMMDD-YYYYMMDD.' will be thrown. It might confuse some people, maybe it will be clearer if you had mentioned something about it in your user guide. 

-------------
Labels: `severity.Low` `type.DocumentationBug`
original: yuxuanxc/ped#2

### LINKED ISSUE #185: [PE-D] TEMPERATURE_RANGE cannot be single digit
If user inputs 'searchpatient tr/0.0-40.0' the error message 'Temperature range should contain temperature-temperature' will be shown, but it was not mentioned that temperature cannot be single digit. ('searchpatient tr/00.0-40.0' works though, so it is a little inconsistent). Maybe can mention the min temperature and max temperature in the error message/user guide?

-------------
Labels: `severity.Low` `type.FeatureFlaw`
original: yuxuanxc/ped#3

### LINKED ISSUE #189: [PE-D] Set clearer boundary for body temperature
When I try to set a patient's body temperature to 99.0, the temperature is being changed sucessfully. However 99.0 is too high for a body temperature. Hence should try setting a boundary for temperature. Eg. 30.0~45.0

-------------
Labels: `severity.Low` `type.FeatureFlaw`
original: yingqi0607/ped#2

### PR DESCRIPTION
**Pull request by chiamyunqing**
Note: Java CI failing due to codecov error (can't do anything about that)
In this PR:
1) Fixed Age class error messages to make it more precise to error
2) Period of stay- 1. allow for one day period of stay (i.e. same day-same day) 2. Update error messages to make it more precise to error
3) Fixed comment bug that allows for empty comment
4) Fixed temperature class - Valid temp is 32.0 to 41.0 (both inclusive)  
5) Update search room error message 
6) Update name class - Valid name is up to 150 char only 
7) Update UI for patient details panel bugs 

I got so many bugs :"(, UG will be updated separately.

### CODE REVIEW

**Comment by chiamyunqing** [comment]
Can only link up to max. 10 issues. Once this PR is closed, to close this following issues too : #208 #188 #193 #201 #158

File: src/main/java/seedu/address/model/patient/Comment.java [hunk 1]
```diff
@@ -17,6 +17,7 @@
      */
     public Comment(String comment) {
         requireNonNull(comment);
+        assert(comment.length() != 0);
```
Comment (w-yeehong) [inline src/main/java/seedu/address/model/patient/Comment.java:20]: Noticed the defensive programming here 👍

File: src/main/java/seedu/address/model/patient/Name.java [hunk 2]
```diff
@@ -35,7 +36,7 @@ public Name(String name) {
      * Returns true if a given string is a valid name.
      */
     public static boolean isValidName(String test) {
-        return test.matches(VALIDATION_REGEX);
+        return test.matches(VALIDATION_REGEX) && test.length() <= 150;
```
Comment (w-yeehong) [inline src/main/java/seedu/address/model/patient/Name.java:39]: An alternative is to check using regex. `[\\p{Alnum}]{1, 150}` should do the trick.

**Review by w-yeehong — APPROVED** [review]
Clean code as usual! LGTM.

**Pull request merged**